std/typecons.d

```diff
@@ -758,34 +758,81 @@ template Tuple(Specs...)
                 h += typeid(T).getHash(cast(const void*)&field[i]);
             return h;
         }
-
-        void toString(DG)(scope DG sink)
+        
+        /**
+            Formats `Tuple` with either `%s`, `%(inner%)` or `%(inner%|sep%)`.
+            
+            `%s` is the original format.
+            `%(inner%)` where `inner` is the format applied the expanded `Tuple`, 
+            so `inner` may contain as many formats as the `Tuple` has fields.
+            `%(inner%|sep%)`, where `inner` is one format, that is applied
+            on all fields of the `Tuple`. The format must be compatible to all
+            of them.
+        */
+        void toString(DG)(scope DG sink, std.format.FormatSpec!char fmt = "%s") const
         {
-            enum header = typeof(this).stringof ~ "(",
-                 footer = ")",
-                 separator = ", ";
-            sink(header);
-            foreach (i, Type; Types)
+            if (fmt.nested)
             {
-                static if (i > 0)
+                import std.format : format;
+                if (fmt.sep)
                 {
-                    sink(separator);
+                    foreach (i, Type; Types)
+                    {
+                        static if (i > 0)
+                        {
+                            sink(fmt.sep);
+                        }
+                        // TODO: Change this once toString() works for shared objects.
+                        static if (is(Type == class) && is(typeof(Type.init) == shared))
```

Comment:
> Just do ``is(Type == shared)``?

I had that in mind, too, but I've never dealt with shared types. Indeed, I believe there is a difference that is not obvious, just because those who wrote that should have had this in mind—but didn't do that. Therefore I left it like that.